### PR TITLE
Deletion: automatically exclude unavailable RSEs #528

### PR DESCRIPTION
**Pull request by tbeerman**

### CODE REVIEW

File: lib/rucio/daemons/reaper/reaper2.py [hunk 1]
```diff
@@ -594,31 +616,33 @@ def stop(signum=None, frame=None):
...
-    :param exclude_rses:       RSE expression to exclude RSEs from the Reaper.
-    :param include_rses:       RSE expression to include RSEs.
-    :param vos:                VOs on which to look for RSEs. Only used in multi-VO mode.
-                               If None, we either use all VOs if run from "def",
-                               or the current VO otherwise.
-    :param delay_seconds:      The delay to query replicas in BEING_DELETED state.
-    :param sleep_time:         Time between two cycles.
+    :param threads:                The total number of workers.
+    :param chunk_size:             The size of chunk for deletion.
+    :param noaccess_threshold:     Number of service unavailable exceptions after which the RSE gets temporarily excluded.
```
Comment (bziemons) [inline lib/rucio/daemons/reaper/reaper2.py]: `noaccess_threshold` doesn't exist as parameter

**Review by bziemons — CHANGES_REQUESTED** [review]

File: lib/rucio/daemons/reaper/reaper2.py (hunk 1 above)
Comment (tbeerman) [reply lib/rucio/daemons/reaper/reaper2.py]: thanks, I missed that one.

**Review by bziemons — DISMISSED** [review]

File: bin/rucio-reaper2 [hunk 2]
```diff
@@ -55,6 +56,10 @@ def get_parser():
                         help='The delay (seconds) to query replicas in BEING_DELETED state.')
     parser.add_argument('--scheme', action="store", default=None, type=str,
                         help='Force the reaper to use a particular protocol/scheme, e.g., mock')
+    parser.add_argument("--auto_exclude_threshold", action="store", default=100, type=int,
```
Comment (bari12) [inline bin/rucio-reaper2:59]: As discussed, we should right away also add a config_get from the config table alternatively, so we can more easily experiment with these values.

**Review by bari12 — CHANGES_REQUESTED** [review]

File: bin/rucio-reaper2 (hunk 2 above)
Comment (tbeerman) [reply bin/rucio-reaper2:59]: Which one should be preferred? CLI argument or config table?

**Review by bari12 — APPROVED** [review]

**Pull request merged**